src/sources/kafka.rs

```diff
@@ -208,18 +197,44 @@ async fn kafka_source(
                         }
                     }
                 }
-                log.insert(&headers_key, Value::from(headers_map));
+
+                let msg_topic = msg.topic().to_string();
+                let msg_partition = msg.partition();
+                let msg_offset = msg.offset();
+
+                let mut events = Vec::new();
+
+                while let Ok(Some((next, _))) = decoder.decode_eof(&mut payload) {
+                    for mut event in next {
```

Comment:
If you want to avoid this `Vec`, you should be able to wrap `payload` in a `FramedRead` and get a stream you can `map`.

Reply:
Thanks @bruceg and @lukesteensen for the suggestions! Went with `FramedRead` + `async_stream::stream` to make it work